server.py

```
@@ -638,233 +767,110 @@
+    # Basic logging and activity tracking
     logger.info(f"MCP tool call: {name}")
     logger.debug(f"MCP tool arguments: {list(arguments.keys())}")
 
-    # Log to activity file for monitoring
     try:
         mcp_activity_logger = logging.getLogger("mcp_activity")
         mcp_activity_logger.info(f"TOOL_CALL: {name} with {len(arguments)} arguments")
-    except Exception:
-        pass
+    except (OSError, AttributeError) as e:
+        logger.warning(f"Failed to log tool call activity: {e}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)